src/v/rpc/test/rpc_gen_cycling_test.cc

```diff
@@ -73,6 +78,123 @@ FIXTURE_TEST(rpcgen_tls_integration, rpc_integration_fixture) {
     BOOST_REQUIRE_EQUAL(ret.value().data.x, 66);
 }
 
+class temporary_dir {
```

Comment:
nit: this may be something to consider adding to a top-level test utilities namespace/folder for use by all tests.

Reply:
nevermind. this looks like it is kind of specialized for the use case.